### ISSUE TITLE
Using Alternative Transport

### ISSUE THREAD

**Issue by wuftymerguftyguff** [swl10/pyslet#75]
Hi,

Firstly thanks so much for all your work.

May I ask a question?

I am using pyslet odata in a project and have been strugging with athentication

The service is behind various auth barriers including requirement for HTTPS client certificates and multiple layers of SAML redirects.

I have a subclass of the python requests library that deals with all the auth issues and I can get the raw xml back myself. (Horribly implementation specific for the SAML)

Have you considered allowing your odata client to optionally use other http back ends?

Thanks

Jeff

**Comment by swl10** [comment]
Yes, I have considered this.  I haven't rejected the idea, requests has largely been developed in parallel over the years and is now a de facto standard.  I think it is time to make this an enhancement request and add in an appropriate wrapper to enable a requests-based client to be used instead.  I've marked it as such, it really shouldn't be a big change as the API to the actual HTTP client I use is fairly minimal.
